vulnfeeds/cmd/cvelist2osv/main.go

```diff
@@ -82,107 +88,66 @@ var VendorProductDenyList = []cves.VendorProduct{
 	{Vendor: "gradle", Product: "enterprise"}, // The OSS repo gets mis-attributed via CVE-2020-15767
 }
 
-// formatDateTimeRFC3339 parses an ISO-like date string and returns RFC3339 format or original on error.
-func formatDateTimeRFC3339(dateStr string) string {
-	if dateStr == "" {
-		return ""
-	}
-	// Try parsing common ISO8601 variations
-	layouts := []string{
-		time.RFC3339Nano,           // "2006-01-02T15:04:05.999999999Z07:00"
-		time.RFC3339,               // "2006-01-02T15:04:05Z07:00"
-		"2006-01-02T15:04:05Z",     // ISO 8601 UTC
-		"2006-01-02T15:04:05.000Z", // Explicit milliseconds UTC
-	}
-	var t time.Time
-	var err error
-	for _, layout := range layouts {
-		t, err = time.Parse(layout, dateStr)
-		if err == nil {
-			break // Parsed successfully
-		}
-	}
-
-	if err != nil {
-		fmt.Printf("Warning: Could not parse date string '%s': %v. Returning original.\n", dateStr, err)
-		return dateStr // Return original if parsing fails
-	}
-	return t.UTC().Format(time.RFC3339) // Format in UTC RFC3339
-}
-
-// BORROWED FROM VULNS.GO
-// FromCVE creates a minimal OSV object from a given CVEItem and id.
-// Leaves affected and version fields empty to be filled in later with AddPkgInfo
-func FromCVE(id cves.CVEID, cve cves.CVE5) (*vulns.Vulnerability, []string) {
-	aliases, related := vulns.ExtractReferencedVulns(id, cves.CVEID(cve.Metadata.CVEID), cve.Containers.CNA.References)
-	v := vulns.Vulnerability{
-		ID:      string(id),
-		Details: cves.EnglishDescription(cve.Containers.CNA.Descriptions),
-		Aliases: aliases,
-		Related: related,
-	}
+// extractVersionsFromAffected extracts affected versions from the CNA container of a CVE.
+// It populates the models.AffectedVersion slice based on the versioning information
+// provided in the CVE data, handling both explicit ranges (lessThan, lessThanOrEqual)
+// and single affected versions. It also attempts to extract versions from text
+// descriptions if direct version ranges are not available.
+//
+// Parameters:
+//   - cna: The CNA container from the CVE5 object.
+//   - gotVersions: A boolean indicating if any versions have already been extracted.
+//   - versionInfo: The models.VersionInfo struct to populate with affected versions.
+//
+// Returns:
+//   - []models.AffectedVersion: The updated slice of affected versions.
+//   - bool: True if any versions were successfully extracted, false otherwise.
+//   - []string: A slice of notes or warnings generated during the extraction process.
+
+func extractVersionsFromAffected(cna cves.CNA, gotVersions bool, versionInfo models.VersionInfo) ([]models.AffectedVersion, bool, []string) {
```

Comment:
If versionInfo is being modified/populated and we need the results, shouldn't it be a pointer?